Tutorial - Build One Productivity Hub app by using Microsoft Graph Toolkit (#17843)
microsoftgraph/microsoft-graph-docs: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md

In this step, you'll design your web app by using Microsoft Graph Toolkit components and style it with CSS.

## Initialize the Login component

> In **index.html** inside the `<body></body>`, create `<div></div>` right under the provider and add the Login component inside the div into your code block below:

Reviewer: My previous comment didn't render properly, so recommenting here. I was sugesting moving the `<div>` bit into the code block.

```suggestion
In **index.html** inside the `<body></body>`, add the following code under the provider.

````html
<div>
  <mgt-login></mgt-login>
</div>
````
```

  ```HTML
  <div class="features">

    <div class="header"><div class="title">
      <h2>One Productivity Hub</h2>
      <div class="row">
        <div class="column"><h3>Calendar events</h3></div>
        <div class="column"><h3>To-do tasks</h3></div>
        <div class="column"><h3>Files</h3></div>
      </div>
    </div></div>

    <div class="row" id="content">
      <div class="column" id="mgt-col"></div>
      <div class="column" id="mgt-col"></div>
      <div class="column" id="mgt-col"></div>
    </div>

  </div>
  ```

### Agenda component

Under div tagged with `class="row"`, add the Agenda component inside the first column div:

  ```HTML
  <mgt-agenda></mgt-agenda>
  ```

### To-do component

Under div tagged with `class="row"`, add the To-do component inside the second column div:

  ```HTML
  <mgt-todo></mgt-todo>
  ```

### FileList component

Under div tagged with `class="row"`, add the File list component inside the third column div:

  ```HTML
  <mgt-file-list></mgt-file-list>
  ```

## Style your web app with CSS

> 1. Create **index.css** file under your project and add below CCS code:

Reviewer: "add the following CSS code" (typo in CCS)

Author: Done

        display: flex;
        background-color: #f0f0f0;
    }
    .title {
        margin-top: 20px;
        margin-left: 10px;
        width: 100%;
    }
    .title h2 {
        font-size: 24px;
        padding-left: 5px;
        display: inline;
        font-weight: 600;    
    }
    .title h3 {
        float: left;
        width: 32%;
        background:transparent;
        font-size: 16px;
        margin-bottom: 10px;
        padding-left: 10px;
        padding-top: 10px;
        color: #8A8886;
        font-weight: 600;
    }
    mgt-login {   
        margin-left: 20px;
        --avatar-size: 60px;
        --font-family: 'Segoe UI';
        --font-size: 20px;
        --font-weight: 700;
        --color: black;
        --text-transform: none;
        --line2-font-size: 14px;
        --line2-font-weight: 400;
        --line2-color: #8A8886;
        --line2-text-transform: none;
    }
    #content, html, body {
        height: 98%; 
      }
    #mgt-col {
      float: left;
      width: 32%;
      background:transparent;
      height:500px;
      overflow: hidden;
      padding: 5px;
      margin-top: 5px;
    }
    #mgt-col:hover {
      overflow-y: auto;
    }
    ```

> 1. In **index.html** under `<head></head>`, define the stylesheet link `href` as **index.css**:

Reviewer: "inside `<head></head>`"

Author: done

In this step, you'll design your web app by using Microsoft Graph Toolkit components and style it with CSS.

## Initialize the Login component

> In **index.html** inside the `<body></body>`, create `<div></div>` right under the provider and add the Login component inside the div:

Reviewer: I'd add the <div> into your code block below.

Author: done

Reviewer: Doh, just realized what I typed there didn't render :D. I meant to add the `<div>` bit into the code block.

Author: commited your suggestion there. Apologies I didn't understand in the first place :)

  ```html

>   <mgt-login></mgt-login>

Reviewer: Can this be simplified as `<mgt-login/>`?

Author: Unfortunately, we need to use the component as `<mgt-login></mgt-login>`. Tested the suggested way, UI doesn't appear.

Reviewer: I was able to get it to work for this and all of the MGT components. The key is there should be a space before the `/`, so `<mgt-login />`
